packaging/requirements.py

```diff
@@ -6,10 +6,14 @@
 import string
 import re
 
+try:
+    from urllib import parse as urlparse
```

Comment:
instead of try/except, why not have an if else and check on python version?

Reply:
With:
```python
if sys.version_info.major >= 3:
    from urllib import parse as urlparse
else:  # pragma: no cover
    import urlparse
```
mypy still complains:
```
mypy.....................................................................Failed
- hook id: mypy
- exit code: 1

packaging/requirements.py:21: error: Name 'urlparse' already defined (by an import)
Found 1 error in 1 file (checked 12 source files)

mypy for Python 2........................................................Failed
- hook id: mypy
- exit code: 1

packaging/requirements.py:19: error: Module 'urllib' has no attribute 'parse'
packaging/requirements.py:21: error: Name 'urlparse' already defined (possibly by an import)
Found 2 errors in 1 file (checked 11 source files)
```

Reply:
Looks like `sys.version_info[0]` should work over `sys.version_info.major`: https://github.com/python/mypy/pull/1942#issuecomment-369780372

Reply:
Thanks, that works! Will update.

Here's the issue to add `sys.version_info.major` in mypy.

Reply:
Thanks @gaborbernat and @rkm for the help here! ^>^